go/vt/tabletserver/query_splitter_test.go

```diff
@@ -314,3 +314,32 @@ func TestSplitQuery(t *testing.T) {
 		t.Errorf("wrong splits, got: %v, want: %v", got, want)
 	}
 }
+
+func TestSplitQueryStringColumn(t *testing.T) {
+	schemaInfo := getSchemaInfo()
+	query := &proto.BoundQuery{
+		Sql: "select * from test_table where count > :count",
+	}
+	splitter := NewQuerySplitter(query, "", 3, schemaInfo)
+	splitter.validateQuery()
+	splits, err := splitter.split(mproto.VT_VAR_STRING, nil)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	got := []string{}
+	for _, split := range splits {
+		if split.RowCount != 89478485 {
+			t.Errorf("wrong RowCount, got: %v, want: %v", split.RowCount, 100)
+		}
+		got = append(got, split.Query.Sql)
+	}
+	want := []string{
+		"select * from test_table where count > :count and id < hex(89478485)",
+		"select * from test_table where count > :count and id >= hex(89478485) and id < hex(178956970)",
```

Comment:
In SplitQuery, we guarantee the split column must be indexed; otherwise, SplitQuery will fail. In addition to your concern, I think we should only do SplitQuery on column with VARBINARY type. On my local test (generating 10,000 random byte strings and apply SplitQuery to get all split points), it shows that mysql will use index.